src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -647,7 +647,7 @@ private List<AzureTenant> ListAccountTenants(AzureAccount account, AzureEnvironm
                 AzureContext context = new AzureContext(_profile.Context.Subscription, account, environment, 
                                             CreateTenantFromString(tenantId, accessToken.TenantId));
 
-                return subscriptionClient.ListAll().Select(s => s.ToAzureSubscription(context));
+                return subscriptionClient.ListAll().Select(s => s.ToAzureSubscription(context)).ToList();
```

Comment:
We shouldn't need to do this, we should be able to return the IEnumerable without enumerating the entire thing.